src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -453,6 +453,64 @@ public function isAbsolutePath($file)
     }
 
     /**
+     * Creates a temporary file with support for custom stream wrappers.
+     *
+     * @param string $dir    The directory where the temporary filename will be created.
+     * @param string $prefix The prefix of the generated temporary filename.
+     *                       Note: Windows uses only the first three characters of prefix.
+     *
+     * @return string The new temporary filename (with path), or FALSE on failure.
+     */
+    public function tempNam($dir, $prefix)
```

Comment:
Pleas also add a line in the CHANGELOG of the component.
I'm not convinced by this implementation. What about:

```php
do {
    $tmpnam = $dir.'/'.$prefix.mt_rand();
} while (file_exists($tmpnam));

return $tmpnam;
```